clang/tools/c-index-test/core_main.cpp

```diff
@@ -779,41 +783,47 @@ static int scanDeps(ArrayRef<const char *> Args, std::string WorkingDirectory,
             LookupOutputCB.Context, LookupOutputCB.Callback,
             /*Options=*/0, &Error);
     if (Result) {
-      Failed = false;
       llvm::outs() << "dependencies:\n";
       HandleCommand(Result->ContextHash, Result->ModuleDeps, Result->FileDeps,
                     Result->BuildArguments);
       clang_experimental_FileDependencies_dispose(Result);
+      return 0;
     }
-  } else {
-    CXFileDependenciesList *Result = nullptr;
-    clang_experimental_DependencyScannerWorker_getFileDependencies_v4(
-        Worker, Args.size(), Args.data(),
-        ModuleName ? ModuleName->c_str() : nullptr, WorkingDirectory.c_str(),
-        CB.Context, CB.Callback, LookupOutputCB.Context,
-        LookupOutputCB.Callback,
-        /*Options=*/0, &Result, &Error);
-    if (Result) {
-      Failed = false;
-      llvm::outs() << "dependencies:\n";
-      for (size_t I = 0; I < Result->NumCommands; ++I)
-        HandleCommand(
-            Result->Commands[I].ContextHash, Result->Commands[I].ModuleDeps,
-            Result->Commands[I].FileDeps, Result->Commands[I].BuildArguments);
-      clang_experimental_FileDependenciesList_dispose(Result);
-    }
-  }
-
-  if (Failed) {
     llvm::errs() << "error: failed to get dependencies\n";
     llvm::errs() << clang_getCString(Error) << "\n";
     clang_disposeString(Error);
     return 1;
   }
 
-  clang_experimental_DependencyScannerWorker_dispose_v0(Worker);
-  clang_experimental_DependencyScannerService_dispose_v0(Service);
-  return 0;
+  CXFileDependenciesList *Result = nullptr;
+  CXDiagnosticSet Diags;
+  CXErrorCode Err =
+      clang_experimental_DependencyScannerWorker_getFileDependencies_v5(
+          Worker, Args.size(), Args.data(),
+          ModuleName ? ModuleName->c_str() : nullptr, WorkingDirectory.c_str(),
+          CB.Context, CB.Callback, LookupOutputCB.Context,
+          LookupOutputCB.Callback,
+          /*Options=*/0, &Result, &Diags);
+  if (Err == CXError_Success) {
+    llvm::outs() << "dependencies:\n";
+    for (size_t I = 0; I < Result->NumCommands; ++I)
+      HandleCommand(
+          Result->Commands[I].ContextHash, Result->Commands[I].ModuleDeps,
+          Result->Commands[I].FileDeps, Result->Commands[I].BuildArguments);
+    clang_disposeDiagnosticSet(Diags);
```

Comment:
Nit: good candidate for another `make_scope_exit`.

Reply:
Nice catch, thanks!